src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,518 @@
+            break;
+        case COMMAND_DEADLINE:
+            addTask(COMMAND_DEADLINE, params);
+            break;
+        case COMMAND_EVENT:
+            addTask(COMMAND_EVENT, params);
+            break;
+        case COMMAND_LIST:
+            printTasks();
+            break;
+        case COMMAND_DONE:
+            markTaskAsDone(params);
+            break;
+        case COMMAND_DELETE:
+            deleteTask(params);
+            break;
+        case COMMAND_BYE:
+            exit();
+            break;
+        case COMMAND_COMMAND_LIST:
+            showListOfCommands();
+            break;
+        default:
+            showMessageFramedWithDivider(MESSAGE_ERROR_COMMAND_DOES_NOT_EXIST, MESSAGE_SUGGEST_COMMAND_LIST);
+            break;
+        }
+    }
+
+    /**
+     * Creates and adds specific type of task to the Task array
+     *
+     * @param typeOfTask Guaranteed to be either COMMAND_TODO, COMMAND_DEADLINE, or COMMAND_EVENT
+     * @param params String containing description and other info of the task
+     */
+    public static void addTask(String typeOfTask, String params) {
+        final String[] descriptionAndInfo;
+        try {
+            descriptionAndInfo = splitParamsIntoDescriptionAndInfo(params);
+            final String description = descriptionAndInfo[0];
+            final String info = descriptionAndInfo[1];
+            Task task;
+            switch (typeOfTask){
+            case COMMAND_TODO:
+                task = new Todo(description);
+                break;
+            case COMMAND_DEADLINE:
+                String dateDeadline = extractDate(COMMAND_DEADLINE_PREFIX, COMMAND_DEADLINE, info);
+                task = new Deadline(description, dateDeadline);
+                break;
+            case COMMAND_EVENT:
+                String dateEvent = extractDate(COMMAND_EVENT_PREFIX, COMMAND_EVENT, info);
+                task = new Event(description, dateEvent);
+                break;
+            default:
+                //Since typeofTask is guaranteed to be either of the 3 above, code should not be reaching this point
+                throw new InvalidCommandFormatException("Error occurred, please check command format!");
+            }
+
+            tasks.add(task);
+            appendNewTaskToFile(task);
+            showMessageFramedWithDivider("Added to list: " , task.toString(),
+                    "Current number of tasks: " + tasks.size());
+        } catch (InvalidCommandFormatException e) {
+            //Print error for any invalid command format exceptions caught
+            showMessageFramedWithDivider(e.toString());
+        }
+    }
+
+    /**
+     * Deletes a specific task
+     * If task number is not provided or invalid, prints an error
+     *
+     * @param params String in the format "delete X", where X is supposed to be the task number
+     */
+    public static void deleteTask(String params) {
+        if (tasks.isEmpty()) {
```

Comment:
Perhaps you can try to [make the happy path prominent here](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent). Similarly for `markTaskAsDone`.